Controller/Adminhtml/IndexingManager/Reindex.php

```diff
@@ -100,6 +101,28 @@ protected function defineRedirectPath(array $params): string
         return $redirect;
     }
 
+    /**
+     * Defines if all entities need to be reindex
+     *
+     * @param array $params
+     * @return bool
+     */
+    protected function isFullIndex(array $params): bool
+    {
+        return isset($params["entity"]) && $params["entity"] === 'all';
+    }
+
+    /**
+     * Check if the request is coming from a grid (products or pages)
+     *
+     * @param array $params
+     * @return bool
+     */
+    protected function isComingFromGrid(array $params): bool
```

Comment:
Minor nitpick but don't _both_ operations come from the grid page? 😅

Reply:
I consider it "from grid" when you select entites directly inside the grid (with the checkboxes) . The full indexing buttons have their own `redirect` param . Do you want something more specific such as `isComingFromGridEntitySelection` maybe ?

Reply:
Yeah the problem is that button is literally _on the grid page_ which is why I suggest `isMassAction()` to distinguish the _selection_ of items via that mechanism. (Less verbose and correlates to the [reference](https://github.com/algolia/algoliasearch-magento-2/pull/1708/files#diff-ec4950d12f961ad8231677fef21d26d228af5166495fe3db651cee0cf63c5362R9) in the corresponding UI Component XML)

Reply:
updated !